### PR TITLE
Add new ChoiceBlock component.

### PR DESCRIPTION
**Pull request by jameslu123**
Added ChoiceBlock component which allows users to pick the add-on and selection they want.

### CODE REVIEW

File: restaurant-app-template/RestaurantApp/App/Components/ChoiceBlock.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      {
+        elems.push(<CheckBox title={name}
+          checkedIcon='dot-circle-o'
+          uncheckedIcon='circle-o'
+          checked={false}
+          onPress={() => {this._addToPreCart(i)}}
+          key={generate()} />);
+      }
+    }
+
```
Comment (ygao76) [inline restaurant-app-template/RestaurantApp/App/Components/ChoiceBlock.js]: renderRow = (rowData) => {
    const { onPressAction } = this.props
    const { selectedItem } = this.state.selectedItem
    return (
      <CheckBox title={rowData.name}
           checkedIcon='dot-circle-o'
           uncheckedIcon='circle-o'
           checked={rowData.name ==selectedItem}
           onPress={(rowData) => { this.setState({ rowData })}}
           key={generate()}/>
    )
  }

  render() {
    const { list } = this.props.choiceData.selection||[]
    return (
      <ListView
        style={styles.container}
        dataSource={list}
        renderRow={this.renderRow}
      />
    )
  }

Comment (jameslu123) [reply restaurant-app-template/RestaurantApp/App/Components/ChoiceBlock.js]: Looks good, I'll use this.

Comment (jameslu123) [reply restaurant-app-template/RestaurantApp/App/Components/ChoiceBlock.js]: FYI I also need to dispatch a redux action in onPress of checkbox.

Comment (jameslu123) [reply restaurant-app-template/RestaurantApp/App/Components/ChoiceBlock.js]: Will address in issue #23

**Pull request merged**